src/lib.rs

```diff
@@ -337,10 +337,8 @@ impl PortOffset {
     /// Creates a new incoming port offset.
     #[inline(always)]
     pub fn new_incoming(offset: usize) -> Self {
-        assert!(offset < u16::MAX as usize);
-        // SAFETY: The value cannot be zero
-        let offset = unsafe { NonZeroU16::new_unchecked(offset.saturating_add(1) as u16) };
-        PortOffset::Incoming(offset)
+        assert!(offset <= u16::MAX as usize);
```

Comment:
Could this and the `as` cast on the next line be combined into `u16::try_from(offset).unwrap()`?

Reply:
That works!